internal/pkg/provider/helm/values_test.go

```diff
@@ -0,0 +1,316 @@
+							"verboseContainerLocatorLogs": false,
+						},
+					},
+				},
+				"spire-server": Values{
+					"caKeyType": "rsa-2048",
+					"caTTL":     "12h",
+					"controllerManager": Values{
+						"enabled": true,
+						"identities": Values{
+							"clusterFederatedTrustDomains": Values{
+								"tz2": Values{
+									"bundleEndpointProfile": Values{
+										"endpointSPIFFEID": "spiffe://td2/spire/server",
+										"type":             "https_spiffe",
+									},
+									"bundleEndpointURL": "127.0.0.2",
+									"trustDomain":       "td2",
+									"trustDomainBundle": "",
+								},
+							},
+							"clusterSPIFFEIDs": Values{
+								"ap1": Values{
+									"federatesWith": ValueList{
+										"td2",
+									},
+									"namespaceSelector": Values{
+										"matchExpressions": ValueList{},
+										"matchLabels": Values{
+											"kubernetes.io/metadata.name": "ns1",
+										},
+									},
+								},
+								"default": Values{
+									"enabled": false,
+								},
+							},
+						},
+					},
+					"federation": Values{
+						"enabled": true,
+					},
+					"fullnameOverride": "spire-server",
+					"logLevel":         "DEBUG",
+					"nodeAttestor": Values{
+						"k8sPsat": Values{
+							"allowedNodeLabelKeys": ValueList{},
+							"allowedPodLabelKeys":  ValueList{},
+							"audience": ValueList{
+								"spire-server",
+							},
+							"enabled": true,
+							"serviceAccountAllowList": ValueList{
+								"spire:spire-agent",
+							},
+						},
+					},
+					"service": Values{
+						"type": "LoadBalancer",
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := defaultConfig()
+			source := newFakeDataSource(t, cfg)
+			g := &HelmValuesGenerator{
+				source:    source,
+				trustZone: tt.trustZone,
+			}
+			got, err := g.GenerateValues()
+			require.Nil(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestHelmValuesGenerator_GenerateValues_failure(t *testing.T) {
+	tests := []struct {
+		name          string
+		trustZone     *trust_zone_proto.TrustZone
+		wantErrString string
+	}{
+		{
+			name: "no trust provider",
+			trustZone: func() *trust_zone_proto.TrustZone {
+				tz := fixtures.TrustZone("tz1")
+				tz.TrustProvider = nil
+				return tz
+			}(),
+			wantErrString: "no trust provider for trust zone tz1",
+		},
+		{
+			name: "invalid trust provider kind",
+			trustZone: func() *trust_zone_proto.TrustZone {
+				tz := fixtures.TrustZone("tz1")
+				tz.TrustProvider.Kind = fixtures.StringPtr("invalid")
+				return tz
+			}(),
+			wantErrString: "an unknown trust provider profile was specified: invalid",
+		},
+		{
+			name: "unknown attestation policy",
+			trustZone: func() *trust_zone_proto.TrustZone {
+				tz := fixtures.TrustZone("tz1")
+				tz.AttestationPolicies[0].Policy = "invalid"
+				return tz
+			}(),
+			wantErrString: "failed to find attestation policy invalid in local config",
```

Comment:
More of a small nitpick but maybe we could rename `invalid` here to be `invalid-ap`? Just for verbosity 😃

Reply:
:+1: